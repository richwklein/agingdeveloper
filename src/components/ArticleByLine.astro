---
/**
 * @name ArticleByLine
 * Component used for rendering the by-lines of a single article.
 *
 * @param {Props} props - The properties of the by-line
 * @return {Function} an Astro component factory function.
 *
 * @example
 * <ArticleByLine author={author} published={published} />
 */
import type { CollectionEntry } from "astro:content"
import LinkInternal from "@components/LinkInternal.astro"
import slugify from "@sindresorhus/slugify"

interface Props {
  author: CollectionEntry<"author">
  published: Date
  class?: string
}
const { author: entry, published, class: className } = Astro.props
const author = entry?.data
---

<small id="article-byline" class:list={["text-base text-primary-dark/85", className]}>
  <LinkInternal to={`/author/${slugify(entry.id)}`} class:list={["mr-4 inline-block underline"]}>
    {author?.name}
  </LinkInternal>
  <time datetime={published.toISOString()}>
    {
      published.toLocaleDateString("en-us", {
        year: "numeric",
        month: "long",
        day: "numeric",
      })
    }
  </time>
</small>
