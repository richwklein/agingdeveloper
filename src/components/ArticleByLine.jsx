import React from "react";
import {Box, Typography} from "@mui/material";
import moment from "moment";
import PropTypes from "prop-types";
import {ArticleAuthorDigestProps} from "../props";
import InternalLink from "./InternalLink";

/**
 * React component for rendering an article's by-line.
 *
 * @param {ArticleByLineProps} props - The article by-line props.
 * @return {React.ReactElement} - The react component
 *
 * @example
 * <ArticleByLine author={author} published={published} />
 */
export const ArticleByLine = ({author, published}) => {
  const formatted = moment.utc(published).format("MMMM DD, YYYY");

  return (
    <Box>
      <InternalLink to={`/author/${author.slug}`} variant="subtitle1"
        sx={{display: "inline-block", p: .5, mr: 1}}>
        {author.name}
      </InternalLink>
      <Typography variant="subtitle2" sx={{display: "inline-block", p: .5}}>
        {formatted}
      </Typography>
    </Box>
  );
};

/**
 * @typedef ArticleByLineProps - Article by-line props
 * @property {ArticleAuthorDigestProps} author - The author of the article.
 * @property {string} published - The date the article was published.
 */
ArticleByLine.propTypes = {
  author: ArticleAuthorDigestProps.isRequired,
  published: PropTypes.string.isRequired,
};

export default ArticleByLine;
