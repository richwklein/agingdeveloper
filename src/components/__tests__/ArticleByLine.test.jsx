import React from "react";
import {cleanup, render} from "@testing-library/react";
import ArticleByLine from "../ArticleByLine";

describe("ArticleByLine", () => {
  afterEach(cleanup);

  const renderComponent = () => render(
      <ArticleByLine
        author={{name: "Georgia Smith", slug: "georgia-smith"}}
        published="2023-11-02T15:30:00Z" />,
  );

  test("that it renders correctly.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("that it has a link to the authors page.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toContainHTML("href=\"/author/georgia-smith\"");
  });
});
