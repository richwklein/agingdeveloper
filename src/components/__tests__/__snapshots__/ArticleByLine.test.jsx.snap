// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleByLine that it renders correctly. 1`] = `
<div
  class="MuiBox-root css-0"
>
  <a
    class="MuiTypography-root MuiTypography-subtitle1 MuiLink-root MuiLink-underlineAlways css-13wm0u9-MuiTypography-root-MuiLink-root"
    data-link="internal"
    href="/author/georgia-smith"
  >
    Georgia Smith
  </a>
  <h6
    class="MuiTypography-root MuiTypography-subtitle2 css-92xwbh-MuiTypography-root"
  >
    November 01, 2023
  </h6>
</div>
`;
